modules/govuk/templates/node/s_graphite/remove_old_whisper_data.erb

```diff
@@ -14,6 +14,14 @@ for dir in $whisper_dir/*; do
     rm -rf $dir;
   fi
 done
+
+for dir in stats stats_counts; do
+  find "$whisper_dir/$dir" -name "*ip-*" -type d -print0 | while IFS= read -r -d '' dir; do
+    if [[ ! -z $dir && -z "$(find $dir -mtime -56)" && $? -eq 0 ]]; then
```

Comment:
Can you help me understand what the `$? -eq 0` bit is for?

Reply:
I'm not entirely certain (this bit is copied from further up in the file), but I think it's just an extra check that the `find` command executed successfully. `$?` is the exit status of the last command, and as far as I know, `find` will exit with 0 regardless of the results. I believe it could be removed without changing the behavior of the script.